engine/function/src/templates/Numeric.ftl

```diff
@@ -1564,29 +1621,36 @@ public class Numeric {
         }
 
         if (count == 0) {
-            return ${pt.null};
+            return NULL_LONG;
         }
 
-    <#if pt.valueType.isFloat >
-        return zeroCount > 0 ? 0 : (${pt.primitive}) (prod);
-    <#else>
         return (${pt.primitive}) (prod);
```

Comment:
Shouldn't this be just `return prod;`?
I would assume, now that we've widened the return type, we *don't* want the narrowing cast here.

Reply:
Good catch